src/experimental/slam_toolbox_map_and_localization.cpp

```diff
@@ -26,21 +26,41 @@ MapAndLocalizationSlamToolbox::MapAndLocalizationSlamToolbox(rclcpp::NodeOptions
 : LocalizationSlamToolbox(options)
 /*****************************************************************************/
 {
+}
+
+/*****************************************************************************/
+void MapAndLocalizationSlamToolbox::configure()
+/*****************************************************************************/
+{
+  SlamToolbox::configure();
+  toggleMode(false);
+}
+
+/*****************************************************************************/
+CallbackReturn
+MapAndLocalizationSlamToolbox::on_configure(const rclcpp_lifecycle::State &)
+/*****************************************************************************/
+{
+  processor_type_ = PROCESS;
+  SlamToolbox::on_configure(rclcpp_lifecycle::State());
   // disable interactive mode
   enable_interactive_mode_ = false;
 
   ssSetLocalizationMode_ = create_service<std_srvs::srv::SetBool>(
     "slam_toolbox/set_localization_mode",
     std::bind(&MapAndLocalizationSlamToolbox::setLocalizationModeCallback, this,
     std::placeholders::_1, std::placeholders::_2, std::placeholders::_3));
+  return CallbackReturn::SUCCESS;
 }
 
 /*****************************************************************************/
-void MapAndLocalizationSlamToolbox::configure()
+CallbackReturn
+MapAndLocalizationSlamToolbox::on_cleanup(const rclcpp_lifecycle::State &)
```

Comment:
Why not populate state in the argument and pass that onto the function instead of creating a default constructed `State()` to call the base class' implementation? Ditto everywhere

Reply:
fixed [3997194b6d329d2dbfa0c1558423f6cd02e5aaed]